zend_abstract_interface/zai_sapi/zai_sapi_io.h

```diff
@@ -0,0 +1,46 @@
+#ifndef ZAI_SAPI_IO_H
+#define ZAI_SAPI_IO_H
+
+#include <stddef.h>
+
+/* Writes to stdout (unbuffered).
+ *
+ * Output buffering is handled by PHP. Once PHP is ready to flush it sends the
+ * output to the SAPI's unbuffered write operation. We do not buffer when
+ * writing to stdout as to not conflict with the PHP output buffer.
+ */
+size_t zai_sapi_io_write_stdout(const char *str, size_t len);
+
+/* Writes to stderr (unbuffered).
+ *
+ * As PHP does not buffer stderr, it is not required to have an unbuffered
+ * write operation for stderr at the SAPI level. But we still perform
+ * unbuffered write operations to stderr for consistency with stdout.
+ */
+size_t zai_sapi_io_write_stderr(const char *str, size_t len);
+
+/* Flushes stdout (NOOP).
+ *
+ * ZAI SAPI only supports unbuffered write operations.
+ */
+static inline void zai_sapi_io_flush(void *server_context) { (void)server_context; }
```

Comment:
Why are we adding this `static` function here in the header? Shouldn't we either remove `static` or have it declared only in the file we use it?

Reply:
If you really want to get rid of `static`, that's fine, it just means in some object (and just one) we add:

```c
extern inline void zai_sapi_io_flush(void *server_context);
```